crucible.mjs

```diff
@@ -70,16 +70,72 @@ Hooks.once("init", async function() {
   /**
    * Configurable properties of the system which affect its behavior.
    */
+  function itemPacksByType(type) {
+      let potentialPacks = {};
+      for (const pack of game.packs) {
+        if (pack.metadata.type !== "Item") continue;
+        for (const item of pack.index) {
+          if (item.type === type) {
+            let name = pack.metadata.label;
+            let folder = pack.folder;
+            while (folder) {
+              name = `${folder.name} -> ${name}`
+              folder = folder.folder;
+            }
+            potentialPacks[pack.metadata.id] = name;
+            break;
+          }
+        }
+      }
+      return potentialPacks;
+  }
+  game.settings.register("crucible", "ancestrySources", {
+    name: "SETTINGS.AncestrySourcesName",
+    hint: "SETTINGS.AncestrySourcesHint",
+    scope: "world",
+    config: true,
+    type: new foundry.data.fields.SetField(new foundry.data.fields.StringField({ required: true, choices: () => itemPacksByType("ancestry")})),
+    default: [SYSTEM.COMPENDIUM_PACKS.ancestry],
+    requiresReload: true,
+  });
+  game.settings.register("crucible", "backgroundSources", {
+    name: "SETTINGS.BackgroundSourcesName",
+    hint: "SETTINGS.BackgroundSourcesHint",
+    scope: "world",
+    config: true,
+    type: new foundry.data.fields.SetField(new foundry.data.fields.StringField({ required: true, choices: () => itemPacksByType("background")})),
+    default: [SYSTEM.COMPENDIUM_PACKS.background],
+    requiresReload: true,
+  });
+  game.settings.register("crucible", "spellSources", {
+    name: "SETTINGS.SpellSourcesName",
+    hint: "SETTINGS.SpellSourcesHint",
+    scope: "world",
+    config: true,
+    type: new foundry.data.fields.SetField(new foundry.data.fields.StringField({ required: true, choices: () => itemPacksByType("spell")})),
+    default: [SYSTEM.COMPENDIUM_PACKS.spell],
+    requiresReload: true,
+  });
+  game.settings.register("crucible", "talentSources", {
+    name: "SETTINGS.TalentSourcesName",
+    hint: "SETTINGS.TalentSourcesHint",
+    scope: "world",
+    config: true,
+    type: new foundry.data.fields.SetField(new foundry.data.fields.StringField({ required: true, choices: () => itemPacksByType("talent")})),
+    default: [SYSTEM.COMPENDIUM_PACKS.talent],
+    requiresReload: true,
+  });
+  
   crucible.CONFIG = {
     /**
      * Configuration of compendium packs which are used as sources for system workflows.
      * @type {Record<string, Set<string>>}
      */
     packs: {
```

Comment:
The `init` hook is too early to retrieve world-level settings. We need to move the definition of `CONFIG.packs` later on into the `"setup"` hook (but before we initialize the talent tree).

Reply:
Just to be clear, we still want it to be in `crucible.CONFIG`, but we want to update it in `"setup"`, right? In that case I probably should initialize the sets as empty and then add to them based on the setting, that way if another module wants to modify the config in `"init"`, it won't be overriden.
Actually, about modules modifying things, I am not sure how a module is supposed to remove default options if we do this.